spec/models/step_by_step_page_spec.rb

```diff
@@ -216,6 +216,27 @@
     end
   end
 
+  describe 'should_check_links?' do
+    it 'should return false if step by step is scheduled' do
+      step_by_step_with_step = create(:step_by_step_page, status: 'scheduled', scheduled_at: 1.day.from_now)
+
+      expect(step_by_step_with_step.should_check_links?).to be false
+    end
+
+    it 'should return false if step by step has not been edited since publish' do
+      publish_time = Time.zone.now
+      step_by_step_with_step = create(:step_by_step_page, status: 'published', draft_updated_at: publish_time, published_at: publish_time)
+
+      expect(step_by_step_with_step.should_check_links?).to be false
+    end
+
+    it 'should return true if step by step is in draft' do
+      step_by_step_with_step = create(:step_by_step_page, status: 'draft')
```

Comment:
The status doesn't need to be set to "draft" as that's the default for this factory.